SALib/analyze/sobol.py

```diff
@@ -87,50 +88,23 @@ def analyze(problem, Y, calc_second_order=True, num_resamples=1000,
 def first_order(A, AB, B):
     # First order estimator following Saltelli et al. 2010 CPC, normalized by
     # sample variance
-    return np.mean(B * (AB - A)) / np.var(np.r_[A, B])
-
-
-def first_order_confidence(A, AB, B, num_resamples, conf_level):
-    s = np.empty(num_resamples)
-    for i in range(num_resamples):
-        r = np.random.randint(len(A), size=len(A))
-        s[i] = first_order(A[r], AB[r], B[r])
-
-    return norm.ppf(0.5 + conf_level / 2) * s.std(ddof=1)
+    return np.mean(B * (AB - A), axis=0) / np.var(np.r_[A, B], axis=0)

```

Comment:
Again looking at ch4 in  Saltelli's book (2008):
![snippet](https://cloud.githubusercontent.com/assets/3727919/7533178/38b0865c-f567-11e4-8dcd-1e51860877c5.png)
could you do a dot product of A and B e.g.
    f2 = np.exp(np.mean(A), 2)
    np.divide(np.dot(A, AB_i) - f2, dp.dot(A, A) - f2)